script.js

```diff
@@ -1,4 +1,339 @@
+var currentWin = 0;
+var currentDraw = 0;
+var gameCount = 0;
+var currentGameMode = "Waiting for user to enter name";
+var userName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //Do this if current game mode is setting user name
+  if (currentGameMode == "Waiting for user to enter name") {
+    //set the name
+    userName = input;
+    console.log(currentGameMode);
+    //once the user enters the name, game continues.
+    currentGameMode = "SPR";
+    return (
+      "Hello " +
+      userName +
+      ", Please enter scissors/paper/rock intot he box above"
+    );
+  }
+  //if user types in reverse
+  if (input == "reverse") {
+    currentGameMode = "reversed SPR";
```

Comment:
currently your reverse game mode is not doing anything other than have a different output